src/simtools/layout/ctao_array_layouts.py

```diff
@@ -0,0 +1,166 @@
+"""Retrieve, merge, and write layouts from CTAO common identifiers repository."""
+
+import logging
+from pathlib import Path
+
+import simtools.utils.general as gen
+from simtools.data_model.metadata_collector import MetadataCollector
+from simtools.data_model.model_data_writer import ModelDataWriter
+from simtools.io_operations import io_handler
+from simtools.utils import names
+
+_logger = logging.getLogger(__name__)
+
+
+def retrieve_array_layouts(site, repository_url, branch_name="main"):
+    """
+    Retrieve array layouts from CTAO common identifiers repository.
+
+    Parameters
+    ----------
+    branch_name : str
+        Repository branch to use for CTAO common identifiers.
+
+    Returns
+    -------
+    dict
+        Array layouts for all CTAO sites.
+    """
+    _logger.info(f"Retrieving array layouts from {repository_url} on branch {branch_name}.")
+
+    if gen.is_url(repository_url):
+        array_element_ids = gen.collect_data_from_http(
+            url=f"{repository_url}/{branch_name}/array-element-ids.json"
+        )
+        sub_arrays = gen.collect_data_from_http(
+            url=f"{repository_url}/{branch_name}/subarray-ids.json"
+        )
+    else:
+        array_element_ids = gen.collect_data_from_file(
+            Path(repository_url) / "array-element-ids.json"
+        )
+        sub_arrays = gen.collect_data_from_file(Path(repository_url) / "subarray-ids.json")
+
+    return _get_layouts_per_site(site, sub_arrays, array_element_ids)
+
+
+def _get_layouts_per_site(site, sub_arrays, array_element_ids):
+    """
+    Get array layouts for all CTAO sites.
+
+    Parameters
+    ----------
+    site : str
+        Site identifier.
+    sub_arrays : dict
+        Sub-array definitions.
+    array_element_ids : dict
+        Array element definitions.
+
+    Returns
+    -------
+    dict
+        Array layouts for all CTAO sites.
+    """
+    layouts_per_site = []
+
+    for array in sub_arrays.get("subarrays", []):
+        elements = []
+        for ids in array.get("array_element_ids", []):
+            element_name = _get_array_element_name(ids, array_element_ids)
+            if names.get_site_from_array_element_name(element_name) != site:
+                break
+            elements.append(element_name)
+        if len(elements) > 0:
```

Comment:
Can we have subarrays with zero elements?

Reply:
The issue is that we get here data from outside of simtools and we don't have control. I therefore added this check to be safe.